community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/rotation/LogRotationImpl.java

```diff
@@ -49,18 +49,21 @@ public boolean rotateLogIfNeeded( LogAppendEvent logAppendEvent ) throws IOExcep
          * doing force (think batching of writes), such that it can't see a bad state of the writer
          * even when rotating underlying channels.
          */
-        synchronized ( logFile )
+        if ( logFile.rotationNeeded() )
```

Comment:
For this double-checked locking to work properly, the `PhysicalLogVersionedStoreChannel.position` field needs to be volatile.